src/compiler/checker.ts

```diff
@@ -7870,7 +7870,7 @@ namespace ts {
                 (resolvedSymbol || symbol).exports!.forEach((s, name) => {
                     const exportedMember = members.get(name)!;
                     if (exportedMember && exportedMember !== s) {
-                        if (s.flags & SymbolFlags.Value) {
+                        if (s.flags & SymbolFlags.Value && exportedMember.flags & SymbolFlags.Value) {

```

Comment:
@weswigham @sandersn when I was looking at this with Eli, this brought up a question as to why/how `exportedType` (defined a few lines above—the type of `require('./y.js')` in the test case) could have non-value members. In `resolveAnonymousTypeMembers`, it copies the exports of the symbol to the type, but of course, the `exports` of the .d.ts module includes a type symbol. It was surprising to me that `resolveStructuredTypeMembers` can return an anonymous object type that has members that are also types. Is that normal?

Reply:
normal-ish -- it can easily happen when you merge a type-only namespace with an object literal:

```ts
var o = { x: 1 }
namespace o { export type y = number }
```

Reply:
I realized I was confusing anonymous type _members_ with anonymous type _properties_, which are guaranteed to be values. With that confusion cleared up, this fix definitely looks correct. The possibility of type members was simply overlooked in the past.